src/figuresClasses.ts

```diff
@@ -1,19 +1,72 @@
-export interface Figure {
-
+enum Shape {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle',
 }
 
-export class Triangle {
+type Color = 'green' | 'red' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape = Shape.triangle;
+
+  constructor(public color: Color,
+    public a: number, public b: number, public c: number) {
+    if (a + b <= c
+      || b + c <= a
+      || c + a <= b) {
+      throw new Error('That is not a triagnle!');
+    }
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side value should be more than 0');
+    }
+  }
+
+  private halfSides = (this.a + this.b + this.c) / 2;
 
+  getArea(): number {
```

Comment:
It's a method property used to calculate something inside of it, the class doesnt need to have this itself
```suggestion
  getArea(): number {
    const halfSides = (this.a + this.b + this.c) / 2;
```